doc/source/reference/c-api/array.rst

```diff
@@ -515,7 +515,8 @@ From other objects
 
     Return an ndarray object from a Python object that exposes the
     :obj:`~numpy.class.__array__` method. The :obj:`~numpy.class.__array__`
-    method can take 0, or 1 argument ``([dtype])``. ``context`` is unused.
+    method can take 0, 1 or 2 arguments ``(dtype=None, copy=None)``.
```

Comment:
Sorry for not flagging this on the first review, can you rephrase this to emphasize that third-party `__array__` implementations *must* take `dtype` and `copy` keyword arguments? Can you also document the meaning of those keyword arguments and explain that third-party code or numpy might break if `__array__` ignores the keywords.

Reply:
Ok! I mentioned there that `dtype` and `copy` are mandatory. In `~numpy.class.__array__` docs (which is linked here) I repeated it and explained the meaning of these arguments.